example-datastream-job/src/main/java/org/apache/flink/connector/prometheus/examples/DataStreamJob.java

```diff
@@ -0,0 +1,140 @@
+ *  this work for additional information regarding copyright ownership.
+ *  The ASF licenses this file to You under the Apache License, Version 2.0
+ *  (the "License"); you may not use this file except in compliance with
+ *  the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.apache.flink.connector.prometheus.examples;
+
+import org.apache.flink.api.java.utils.ParameterTool;
+import org.apache.flink.configuration.ConfigOptions;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.connector.base.sink.AsyncSinkBase;
+import org.apache.flink.connector.prometheus.sink.PrometheusRequestSigner;
+import org.apache.flink.connector.prometheus.sink.PrometheusSink;
+import org.apache.flink.connector.prometheus.sink.PrometheusTimeSeries;
+import org.apache.flink.connector.prometheus.sink.PrometheusTimeSeriesLabelsAndMetricNameKeySelector;
+import org.apache.flink.connector.prometheus.sink.aws.AmazonManagedPrometheusWriteRequestSigner;
+import org.apache.flink.connector.prometheus.sink.errorhandling.OnErrorBehavior;
+import org.apache.flink.connector.prometheus.sink.errorhandling.SinkWriterErrorHandlingBehaviorConfiguration;
+import org.apache.flink.connector.prometheus.sink.http.RetryConfiguration;
+import org.apache.flink.connector.prometheus.sink.prometheus.Types;
+import org.apache.flink.streaming.api.datastream.DataStream;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.streaming.api.functions.source.SourceFunction;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Arrays;
+import java.util.function.Supplier;
+
+/** Test application testing the Prometheus sink connector. */
+public class DataStreamJob {
+    private static final Logger LOGGER = LoggerFactory.getLogger(DataStreamJob.class);
+
+    public static void main(String[] args) throws Exception {
+        StreamExecutionEnvironment env;
+
+        // Conditionally return a local execution environment with
+        if (args.length > 0 && Arrays.stream(args).anyMatch("--webUI"::equalsIgnoreCase)) {
+            Configuration conf = new Configuration();
+            conf.set(
+                    ConfigOptions.key("rest.flamegraph.enabled").booleanType().noDefaultValue(),
+                    true);
+            env = StreamExecutionEnvironment.createLocalEnvironmentWithWebUI(conf);
+        } else {
+            env = StreamExecutionEnvironment.getExecutionEnvironment();
+        }
+
+        env.setParallelism(2);
```

Comment:
Can we explain why this is set to 2?

Reply:
Added a comment.
The goal is to force parallel writes and demonstrate how `PrometheusTimeSeriesLabelsAndMetricNameKeySelector` partitions data preventing out-of-order writes, that would be rejected by Prometheus